src/solver/simulation/sim_alloc_probleme_hebdo.cpp

```diff
@@ -59,9 +59,15 @@ void SIM_AllocationProblemeHebdo(PROBLEME_HEBDO& problem, int NombreDePasDeTemps
         problem.BruitSurCoutHydraulique[p] = (double*)MemAlloc(8784 * sizeof(double));
 
     problem.NomsDesPays = (const char**)MemAlloc(nbPays * sizeof(char*));
+    problem.AreaAdequacyPatchMode
+      = (Data::AdequacyPatch::AdequacyPatchMode*)MemAlloc(nbPays * sizeof(int));
```

Comment:
To be placed in the constructor of `AdequacyPatchRuntimeData`
```cpp
AdequacyPatchRuntimeData::AdequacyPatchRuntimeData(uint nbCountries, uint nbLinks) {
/// ...
}
```
You may use `std::vector<AdequacyPatchMode>` instead of `int*` and leave memory management to the standard library.

Reply:
```cpp
class AdequacyPatchRuntimeData
{
private:
   std::vector<double> AreaMode;
   std::vector<int> StartAreaAdequacyPatchType;
   /// ... etc
public:
   void initialize(uint nbCountries, uint nbLinks);
};
```
In `PROBLEME_HEBDO`, add another member of this type.

Here, call 
```cpp
if(adqPatch)
probleme.adqPatchRuntimeData.initialize(nbPays, linkCount);
```